009-ui-and-admin-server-security.md

```diff
@@ -0,0 +1,84 @@
+From thereon, the Kafka authentication and authorization mechanism works in an identical way to the description above for the Kafka Clients.
+
+This mechanism should be straightforward to setup using a `passport.js` strategy.
+The Strimzi UI is served from an Express server and that Express server will be inside the Strimzi cluster allowing the Express Server to talk to the Zookeeper on the secure port 2181.
+The login would not go through the Admin Server which hides the process from the external network but it may be desirable to add a `login` endpoint to the Admin Server which would allow session based applications to use the Admin Server in a natural way.
+
+The Strimzi Admin Server API could be called directly by the user creating an auth header on the request.
+Like the UI flow, the credentials would be retrieved from the auth header and placed into the Kafka client properties as SCRAM credentials.
+
+## OAuth 2
+
+![Kafka Client using Oauth2](./images/oauth-client.png)
+
+In the OAuth 2 scenario, Strimzi currently requires the user to configure the Kafka client with a clientId and secret.
+The client will then obtain an access token from the identity provider which is KeyCloak and that token flows to the broker.
+In the broker, calls to KeyCloak will be made based on the type of token that was requested by the client and passed to the broker.
+
+![Strimzi UI using Oauth2](./images/oauth-ui.png)
+
+In the UI model, the user will be presented with a login dialogue box when the UI is loaded into the browser and requires the user to enter a userId and password.
+The userId needs to be pre-defined to KeyCloak through the KeyCloak user interface.
+When the user enters their userId and password, the UI will contact KeyCloak to authorise the user and obtain an access token.
+This process will be controlled by `passport.js`.
+KeyCloak will either return an access token or reject the request.
+If an access token is returned, it will be stored in the session data for the session.
+When the UI requires data from the Admin Server, the token is recovered from the session data and placed as an auth header on the request.
+The Admin Server receives the request and configures the Kafka client in preparation for calling the Kafka broker.
+The token is retrieved from the auth header and is added to the Kafka client JAAS configuration properties before executing the call to the Kafka broker.
+The Kafka broker, which is running a custom SASL plugin, retrieves the token and sets up the Kafka Principal for the user.
+The request then passes through the KeyCloak Authorizer for authorization.
+
+In order to call the Admin Server directly, the user would need to obtain a token from KeyCloak manually and then store that token in an auth header on the call to the Admin Server.
+The behaviour when the AdminServer received the request would be identical to the flow for the UI calling the Admin Server with a token.
```

Comment:
TBH I read this and I look at the diagram and I think it would be an awful lot simpler if the words referred to the diagram. Even better if the diagram was ascii art because then it would be inline in the text.

````
 OAuthServer  <---------------------------------------+
      ^                                               |
   1. |                                               | 4.
      |        2.                         3.          |
Strimzi UI  -------> Strimzi Admin --------------> Broker

1. UI gets access token as part of OAuth login
2. UI propgates access token via `Xyz` HTTP header
3. Strimzi Admin uses provided token to configure a client
4. Broker validates access token with OAuth server
```